doc/manual/documentation.rst

```diff
@@ -18,25 +24,67 @@ In 0.4:
 
 Documentation is interpreted as `Markdown <https://en.wikipedia.org/wiki/Markdown>`_,
 so you can use indentation and code fences to delimit code examples from text.
+Technically, any object can be associated with any other as metadata;
+Markdown happens to be the default, but one can construct other string
+macros and pass them to the ``@doc`` macro just as well.
+
+Here is a more complex example, still using Markdown:
 
 .. doctest::
 
     """
-    The `@bar` macro will probably make your code 2x faster or something. Use
-    it like this:
+        bar(x[, y])
+
+    Computes the Bar index between `x` and `y`. If `y` is missing, the function computes
+    the index between all pairs of columns of `x`.
 
-        @bar buy_drink_for("Jiahao")
```

Comment:
Yeah, sorry, I wanted a more complex example with optional arguments and a detailed description of the parameters. Maybe add an optional vector of beer types?

More seriously, I realize it would make sense to add the types to the signature, since that's what it advised below.